guides/micronaut-data-r2dbc-repository/java/src/main/java/example/micronaut/GenreController.java

```diff
@@ -0,0 +1,86 @@
+package example.micronaut;
+
+import java.net.URI;
+import java.util.List;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotBlank;
+
+import example.micronaut.domain.Genre;
+import io.micronaut.data.model.Page;
+import io.micronaut.data.model.Pageable;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.annotation.Body;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Put;
+import io.micronaut.http.annotation.Status;
+import reactor.core.publisher.Mono;
+
+@Controller("/genres")  // <1>
+public class GenreController {
+
+    protected final GenreRepository genreRepository;
+
+    public GenreController(GenreRepository genreRepository) { // <2>
+        this.genreRepository = genreRepository;
+    }
+
+    @Get("/{id}") // <3>
+    public Mono<Genre> show(Long id) {
+        return genreRepository
+                .findById(id); // <4>
+    }
+
+    @Put // <5>
+    public Mono<HttpResponse<?>> update(@Body @Valid GenreUpdateCommand command) { // <6>
+        return genreRepository.update(command.getId(), command.getName())
+                    .thenReturn(HttpResponse
+                        .noContent()
+                        .header(HttpHeaders.LOCATION, location(command.getId()).getPath()));  // <7>
+    }
+
+    @Get("/list") // <8>
+    public Mono<List<Genre>> list(@Valid Pageable pageable) { // <9>
+        return genreRepository.findAll(pageable)
+                    .map(Page::getContent);
+    }
+
+    @Post // <10>    
```

Comment:
the trailing whitespace is causing the callout to be rendered as a comment and generating a build warning
```suggestion
    @Post // <10>
```